src/SixLabors.Fonts/Tables/General/CMapTable.cs

```diff
@@ -13,46 +13,29 @@ namespace SixLabors.Fonts.Tables.General
     internal sealed class CMapTable : Table
     {
         internal const string TableName = "cmap";
-        private readonly CMapSubTable[] platformTables;
 
-        public CMapTable(CMapSubTable[] tables)
+        private static readonly Dictionary<PlatformIDs, int> PreferredPlatformOrder = new Dictionary<PlatformIDs, int>
         {
-            this.Tables = tables;
-
-            // lets just pick the best table for us.. lets just treat everything as windows and get the format 4 if possible
-            var tbls = new List<CMapSubTable>();
-            foreach (CMapSubTable t in this.Tables)
-            {
-                if (t != null)
-                {
-                    if (t.Platform == PlatformIDs.Windows)
-                    {
-                        tbls.Add(t);
-                    }
-                }
-            }
+            [PlatformIDs.Windows] = 0,
+            [PlatformIDs.Unicode] = 1,
+            [PlatformIDs.Macintosh] = 2,
+        };
 
-            this.platformTables = tbls.ToArray();
+        public CMapTable(IEnumerable<CMapSubTable> tables)
+        {
+            this.Tables = tables;
         }
 
-        internal CMapSubTable[] Tables { get; }
+        internal IEnumerable<CMapSubTable> Tables { get; }
+
+        private static int GetPreferredPlatformOrder(PlatformIDs platform)
+            => PreferredPlatformOrder.TryGetValue(platform, out var order) ? order : int.MaxValue;
 
         public bool TryGetGlyphId(CodePoint codePoint, out ushort glyphId)
         {
-            // use the best match only
-            foreach (CMapSubTable t in this.platformTables)
-            {
-                // keep looking until we have an index thats not the fallback.
-                if (t.TryGetGlyphId(codePoint, out glyphId))
-                {
-                    return true;
-                }
-            }
-
-            // didn't have a windows match just use any and hope for the best
-            foreach (CMapSubTable t in this.Tables)
+            foreach (CMapSubTable t in this.Tables.OrderBy(t => GetPreferredPlatformOrder(t.Platform)))
```

Comment:
we will want to do the sort once in the constructor rather than on every glyph lookup.

Reply:
Good catch! Addressed in ab083e77a695947c7ec20719ad6c46abb6c3837f.